arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -13,3 +13,18 @@ config ARM_MPU
 	default n
 	help
 	  MCU has ARM MPU
+
+config NXP_MPU
+	bool "NXP MPU Support"
+	depends on CPU_HAS_MPU
+	depends on SOC_FAMILY_KINETIS
+	default n
+	help
+	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
```

Comment:
we should separate this out into a different patch, since I don't see why we need a config MPU_ALLOW_FLASH_WRITE, can't we just do this if CONFIG_FLASH is enabled in the code?  Since that would imply you have the ability to write to the flash